### ISSUE TITLE
Support index directives in Search

### ISSUE THREAD

**Issue by marxin** [sphinx-doc/sphinx#10255]
Right now one can't search for a specific `index` directive in Search box:

```rst
.. index:: my index entry

Indices and tables
==================

* :ref:`genindex`
* :ref:`modindex`
* :ref:`search`
```

It's correctly put to `Index`:
![Screenshot from 2022-03-11 13-52-50](https://user-images.githubusercontent.com/2658545/157870227-215ec79d-254d-424f-b6a2-cca1f58fbf74.png)

But one can't search for it:
![Screenshot from 2022-03-11 13-52-56](https://user-images.githubusercontent.com/2658545/157870249-6c50ac73-b3e7-4fae-8f00-78a9de54f949.png)

**Comment by CAM-Gerlach** [comment]
@AA-Turner This was solved in your PR #10819 landed in 5.2.0, right?

**Comment by marxin** [comment]
Yes, it's implemented, I can confirm that!
